File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/help/HelpSectionFragment.java
package za.co.woolworths.financial.services.android.ui.fragments.help;


import android.app.Fragment;
import androidx.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.os.Bundle;
import androidx.annotation.Nullable;
import android.view.View;
import android.widget.RelativeLayout;

import com.awfs.coordination.BR;
import com.awfs.coordination.R;
import com.awfs.coordination.databinding.NeedHelpFragmentBinding;

import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties;
import za.co.woolworths.financial.services.android.ui.activities.TipsAndTricksViewPagerActivity;
import za.co.woolworths.financial.services.android.ui.base.BaseFragment;
import za.co.woolworths.financial.services.android.ui.fragments.faq.FAQFragment;
import za.co.woolworths.financial.services.android.ui.fragments.help.tipstricks.TipsAndTricksFragment;
import za.co.woolworths.financial.services.android.util.Utils;

/**
 * A simple {@link Fragment} subclass.
 */
public class HelpSectionFragment extends BaseFragment<NeedHelpFragmentBinding, NeedHelpViewModel> implements NeedHelpNavigator, View.OnClickListener {

    RelativeLayout btnTipsAndTricks;
    RelativeLayout btnFAQ;
    NeedHelpViewModel needHelpViewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        needHelpViewModel = ViewModelProviders.of(this).get(NeedHelpViewModel.class);
        needHelpViewModel.setNavigator(this);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        showBackNavigationIcon(true);
        setToolbarBackgroundDrawable(R.drawable.appbar_background);
        setTitle(getString(R.string.need_help));
        showToolbar();
        getViewDataBinding().relFAQ.setOnClickListener(this);
        getViewDataBinding().tipsAndTricks.setOnClickListener(this);
    }

    @Override
    public void onResume() {
        super.onResume();
        Utils.setScreenName(getActivity(), FirebaseManagerAnalyticsProperties.ScreenNames.HELP_SECTION);
    }

    @Override
    public NeedHelpViewModel getViewModel() {
        return needHelpViewModel;
    }


    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.need_help_fragment;
    }

    @Override
    public void onClick(View v) {

        switch (v.getId()) {
            case R.id.relFAQ:
                pushFragment(new FAQFragment());
                break;
            case R.id.tipsAndTricks:
                TipsAndTricksFragment tipsAndTricksFragment = new TipsAndTricksFragment();
                Bundle bundle = this.getArguments();
                if(bundle!=null)
                    tipsAndTricksFragment.setArguments(bundle);
                pushFragment(tipsAndTricksFragment);
                break;
            default:
                break;
        }

    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
        if (!hidden) {
            showBackNavigationIcon(true);
            setToolbarBackgroundDrawable(R.drawable.appbar_background);
            setTitle(getString(R.string.need_help));
            showToolbar();
        }
    }
}

File: app/src/main/java/za/co/woolworths/financial/services/android/ui/fragments/help/HelpSectionFragment.kt
package za.co.woolworths.financial.services.android.ui.fragments.help

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.View.GONE
import com.awfs.coordination.R
import com.awfs.coordination.databinding.NeedHelpFragmentBinding
import za.co.woolworths.financial.services.android.contracts.FirebaseManagerAnalyticsProperties
import za.co.woolworths.financial.services.android.ui.activities.account.MyAccountActivity
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigationActivity
import za.co.woolworths.financial.services.android.ui.activities.dashboard.BottomNavigator
import za.co.woolworths.financial.services.android.ui.fragments.faq.FAQFragment
import za.co.woolworths.financial.services.android.util.Utils
import za.co.woolworths.financial.services.android.util.binding.BaseFragmentBinding

class HelpSectionFragment : BaseFragmentBinding<NeedHelpFragmentBinding>(
    NeedHelpFragmentBinding::inflate
), View.OnClickListener {

    var mBottomNavigator: BottomNavigator? = null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is BottomNavigationActivity)
            mBottomNavigator = context
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupToolbar()

        if (activity is MyAccountActivity) {
            binding.tipsAndTricks.visibility = GONE
        }

        binding.relFAQ.setOnClickListener(this)
        binding.tipsAndTricks.setOnClickListener(this)

    }

    override fun onResume() {
        super.onResume()
        activity?.let {
            Utils.setScreenName(
                it,
                FirebaseManagerAnalyticsProperties.ScreenNames.HELP_SECTION
            )
        }
    }


    override fun onClick(v: View) {
        when (v.id) {
            R.id.relFAQ -> if (activity is BottomNavigationActivity) {
                mBottomNavigator?.pushFragment(FAQFragment())
            } else {
                (activity as? MyAccountActivity)?.replaceFragment(FAQFragment())
            }
            R.id.tipsAndTricks -> {
                val tipsAndTricksFragment = TipsAndTricksFragment()
                val bundle = this.arguments
                if (bundle != null) tipsAndTricksFragment.arguments = bundle
                if (activity is BottomNavigationActivity) {
                    mBottomNavigator?.pushFragment(tipsAndTricksFragment)
                } else {
                    (activity as? MyAccountActivity)?.replaceFragment(tipsAndTricksFragment)
                }
            }
            else -> return
        }
    }

    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)
        if (!hidden)
            setupToolbar()
    }

    private fun setupToolbar() {
        (activity as? BottomNavigationActivity)?.apply {
            showBackNavigationIcon(true)
            setToolbarBackgroundDrawable(R.drawable.appbar_background)
            setTitle(getString(R.string.need_help))
            setToolbarContentDescription(getString(R.string.toolbar_text))
            showToolbar()
        }

        if (activity is MyAccountActivity) {
            (activity as? MyAccountActivity)?.apply {
                supportActionBar?.show()
                setToolbarTitle(activity?.resources?.getString(R.string.need_help))
                setToolbarContentDescription(getString(R.string.toolbar_text))
            }
        }
    }
}